File: app/src/main/java/com/prueba/mytodolist/addTask/AddTaskViewModelFactory.java
package com.prueba.mytodolist.addTask;

import android.app.Application;
import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;

public class AddTaskViewModelFactory extends ViewModelProvider.NewInstanceFactory {

    private final int mTaskId;
    private final Application mApplication;

    public AddTaskViewModelFactory(int taskId,  Application application) {
        mTaskId = taskId;
        mApplication = application;
    }

    @Override
    public <T extends ViewModel> T create(Class<T> modelClass) {
        return (T) new AddTaskViewModel(mTaskId, mApplication);
    }
}

File: app/src/main/java/com/prueba/mytodolist/addTask/AddTaskViewModelFactory.kt
package com.prueba.mytodolist.addTask

import android.app.Application
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class AddTaskViewModelFactory(taskId: Int, application: Application): ViewModelProvider.NewInstanceFactory() {

    private var mTaskId: Int = 0
    private var mApplication: Application

    init {
        mTaskId = taskId
        mApplication = application
    }

    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return AddTaskViewModel(mTaskId, mApplication) as T
    }
}